Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM alyptik/lind:prebuiltsdk
+LABEL lind "v1.0-rc4"
+LABEL description "Lind NaCl Runtime (Pre-built)"
+MAINTAINER Joey Pabalinas <joeypabalinas@gmail.com>
+
+ENV LIND_PREFIX "/home/lind"
+ENV LIND_BASE "$LIND_PREFIX/lind_project"
+ENV LIND_SRC "$LIND_BASE/lind"
+ENV LIND_MONITOR "$LIND_BASE/reference_monitor"
+ENV REPY_PATH "$LIND_SRC/repy"
+ENV NACL_SDK_ROOT "$REPY_PATH/sdk"
+ENV PYTHON "python2"
+ENV PNACLPYTHON "python2"
+ENV LD_LIBRARY_PATH "/lib/glibc"
+ENV PATH "$NACL_SDK_ROOT/toolchain/linux_x86_glibc/bin:$PATH"
+ENV PATH "$REPY_PATH/bin:$PATH"
+ENV PATH "$LIND_BASE:$PATH"
+ENV PATH "/root/bin:/home/lind/bin:$PATH"
+ENV PATH "/root/.local/bin:/home/lind/.local/bin:$PATH"
+
+USER lind
+
+WORKDIR /home/lind/lind_project/
+RUN ["./mklind", "-qe", "nacl"]
```

Comment:
So this is where you pull in the submodule of `native_client` into the base container, right? Are you sure this is happening? The [travis log](https://travis-ci.org/Lind-Project/native_client/builds/450379081#L796-L827) does not look like it. Maybe the docker image `alyptik/lind:prebuiltsdk` is not up to date?

Also, as pointed out in https://github.com/Lind-Project/native_client/issues/1#issuecomment-435068998. If you want to trigger a build that pulls in changes that you pushed to any branch on `native_client` you need to instruct `mklind` to update the submodule to that commit/branch. Currently, you're always updating to the tip of the `develop` branch (see [`git submodule --remote...`](https://git-scm.com/docs/git-submodule/2.0.0#git-submodule---remote) docs and your [`.gitmodules`](https://github.com/Lind-Project/lind_project/blob/master/.gitmodules#L38) `lind_project` branch config for `native_client`).

Maybe you can pass on the branch name from travis environment variables to `mklind` (see [travis docs](https://docs.travis-ci.com/user/environment-variables#convenience-variables) and this [related blog post](https://graysonkoonce.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/))?

Reply:
Not directly related to this PR, but is the [`git clone` fallback in `mklind`](https://github.com/Lind-Project/lind_project/blob/master/mklind#L238-L244) necessary? And is it really only fallback, or does it also interfere with the `git submodule update` call above?

Reply:
This is actually literally only building `nacl` and `repy` in this `Dockerfile`. I split it up into three, one downloads and checks out all the submodules, one builds glibc (this one doesn't pass Travis CI yet), and this one just does the final bits. Even with only these step I am very close to the 50min limit of Travis CI, so I can't really see any other way to do this.

Reply:
Ah, I should probably remove that `# fallback` comment, as it's not really a fallback at all; it's there to set up the submodule symlinks correctly.